pghoard/basebackup.py

```diff
@@ -412,20 +410,123 @@ def add_entry(archive_path, local_path, *, missing_ok):
+        return False
+
+    def create_and_upload_chunks(self, chunks, data_file_format, temp_base_dir):
+        start_time = time.monotonic()
+        chunk_files = []
+        upload_results = []
+        chunk_callback_queue = Queue()
+        chunks_on_disk = 0
+        i = 0
+
+        while i < len(chunks):
+            if chunks_on_disk < self.config["backup_sites"][self.site].get("basebackup_chunk_count", 5):
+                chunk_id = i + 1
+                one_chunk_files = chunks[i]
+
+                chunk_name, input_size, result_size = self.tar_one_file(
+                    callback_queue=chunk_callback_queue,
+                    chunk_path=data_file_format(chunk_id),
+                    temp_dir=temp_base_dir,
+                    files_to_backup=one_chunk_files,
+                )
+                chunk_files.append(
+                    {
+                        "chunk_filename": chunk_name,
+                        "input_size": input_size,
+                        "result_size": result_size,
+                        "files": [chunk[0] for chunk in one_chunk_files]
+                    }
+                )
+                chunks_on_disk += 1
+                i += 1
+                self.log.info("Queued backup chunk %r for transfer, chunks_on_disk: %r, current: %r, total_chunks: %r",
+                              chunk_name, chunks_on_disk, i, len(chunks))
+            elif self.wait_for_chunk_transfer_to_complete(len(chunks), upload_results, chunk_callback_queue, start_time):
+                chunks_on_disk -= 1
+
+        while len(upload_results) < len(chunk_files):
+            self.wait_for_chunk_transfer_to_complete(len(chunks), upload_results, chunk_callback_queue, start_time)
+
+        return chunk_files
+
     def run_local_tar_basebackup(self):
         pgdata = self.config["backup_sites"][self.site]["pg_data_directory"]
         if not os.path.isdir(pgdata):
             raise errors.InvalidConfigurationError("pg_data_directory {!r} does not exist".format(pgdata))
 
-        temp_basebackup_dir, compressed_basebackup = self.get_paths_for_backup(self.basebackup_path)
-
-        compression_algorithm = self.config["compression"]["algorithm"]
-        compression_level = self.config["compression"]["level"]
+        temp_base_dir, compressed_base = self.get_paths_for_backup(self.basebackup_path)
+        os.makedirs(compressed_base)
+        data_file_format = "{}/{}.{{0:08d}}.pghoard".format(compressed_base, os.path.basename(compressed_base)).format
 
-        rsa_public_key = None
-        encryption_key_id = self.config["backup_sites"][self.site]["encryption_key_id"]
-        if encryption_key_id:
-            rsa_public_key = self.config["backup_sites"][self.site]["encryption_keys"][encryption_key_id]["public"]
+        # Default to 2GB chunks of uncompressed data
+        target_chunk_size = self.config["backup_sites"][self.site].get("basebackup_chunk_size") or (1024 * 1024 * 1024 * 2)
```

Comment:
We've been trying to set the default values in `config.py`, please do so for this value, too.